Modules/documents/Controller/DocumentslistController.php

```diff
@@ -101,16 +101,19 @@ public function openAction($id_space, $id) {
         $model = new Document();
         $file = $model->getUrl($id_space,$id);
         
-        header("Cache-Control: public");
-        header("Content-Description: File Transfer");
-        header('Content-Disposition: attachment; filename="'.basename($file).'"' );
-        header("Content-Type: application/zip");
-        header("Content-Transfer-Encoding: binary");
-
-        // read the file from disk
-        readfile($file);
-        //    return;
-        //$this->redirect($path);
+        if ($file && $file != "") {
```

Comment:
If no file, what will occur?
No exception nor view i think

Why not just a if no file raise exception

Reply:
it's more user friendly to have a flash message instead of an exception display... I'm gonna replace the condition with a file_exist($file) and add a try / catch raising an exception

Reply:
flash is fine for me

Reply:
I missed the else condition

Reply:
ok... I just change the condition though and push back